--- sugar-0.86.3/bin/sugar-session.orig	2009-10-20 19:29:27.000000000 +0000
+++ sugar-0.86.3/bin/sugar-session	2010-01-25 21:09:04.000000000 +0000
@@ -213,7 +213,7 @@
 
     client = gconf.client_get_default()
     client.set_string('/apps/metacity/general/mouse_button_modifier',
-                      'disabled')
+                      '<Super>')
 
     timezone = client.get_string('/desktop/sugar/date/timezone')
     if timezone is not None and timezone:
